src/deploymentTemplate.ts

```diff
@@ -0,0 +1,149 @@
+
+export class DeploymentTemplate {
+    constructor() {}
+
+    public displayDeploymentTemplateMenu() {
+
+        TelemetryClient.sendEvent('deploymenttemplate');
+        // currently only one option is available, so first menu won't be displayed yet
+        this.selectQuickstartTemplate();
+    }
+
+    public selectQuickstartTemplate() {
+
+        // get list of directories from here:
+        // https://api.github.com/[repo_user/repo_name]/contents/
+        // we will use fixed repo in our first release
+        let repo: string = "Azure/azure-quickstart-templates";
+
+        var http = require('https');
+        let __this = this;
+
+            http.get({
+                host: Constants.GitHubApiHost,
+                path: '/repos/' + repo + '/contents/',
+                headers: { 'User-Agent': 'VSC Ansible Extension'}
+            }, function(response) {
+                // Continuously update stream with data
+
+                if (response.statusMessage == "OK") {
+                    var body = '';
+                    response.on('data', function(d) {
+                        body += d;
+                    });
+                    response.on('end', function() {
+                        var parsed = JSON.parse(body);
+                        let items: vscode.QuickPickItem[] = [];
+                
+                        for (var i in parsed)
+                        {
+                            // list only directories and skip known directories that don't contain templates
+                            if (parsed[i].type == "dir" && !parsed[i].name.startsWith('.')) {
+                                items.push({ label: parsed[i].name, description: null });
+                            }
+                        }
+                        vscode.window.showQuickPick(items).then(selection => {
+                            // the user canceled the selection
+                            if (!selection) {
+                                return;
+                            }        
+                            
+                            __this.retrieveTemplate(selection.label);
+                        });
+                    });
+                } else {
+                    vscode.window.showErrorMessage("Error: " + response.statusCode + " " + response.statusMessage);
+                }
+
+            }).on('error', function(e) {
+                vscode.window.showErrorMessage("Error: " + e);
+            });
+        
+    }
+
+    public retrieveTemplate(templateName: string) {
+        var http = require('https');
+        let __this = this;
+        let repo: string = utilities.getCodeConfiguration<string>(null, Constants.Config_deploymentTemplatesGitHubRepo);
+
+            http.get({
+                host: Constants.GitHubRawContentHost,
+                path: '/' + repo + '/master/' + templateName + '/azuredeploy.json',
+                headers: { 'User-Agent': 'VSC Ansible Extension'}
```

Comment:
no need to set this user agent?

Reply:
yes, field has to be set to something, otherwise GitHub api will reject.